chocolatey/choco
# Check licensed assembly version during install

## Description Of Changes

Added a check into the chocolateyInstall.ps1 for Chocolatey to check the metadata on the `chocolatey.extension` assembly if it is currently installed/loaded and emit warnings if there is good reason to suspect the assembly may not be fully compatible with the version of Chocolatey being installed.

/cc @gep13 

NOTE: still in draft as there is still some testing to be done to verify this all works as expected. I hope to have that completed Monday morning.

## Motivation and Context

Helps to avoid issues with incompatible licensed extension versions, and give users some resources to help resolve possible problems if they run into issues.

## Testing

1. Run the build.bat and copy the generated chocolatey nupkg to a c:\packages folder
2. Install v1.0.1 of Chocolatey from the community repository.
3. Install a version 3.x.x of `chocolatey.extension`
4. Install the newly built version of Chocolatey from the package in `c:\packages`.
5. You should see a yellow warning message emitted during the installation stating that the extension version is not known to be compatible with the version of Chocolatey being installed.
6. The installation should succeed.
7. Install version 4.0.0 of the `chocolatey.extension` package.
8. Reinstall the prerelease version of Chocolatey from the package in `c:\packages` with `--force` to force the reinstall.
9. No warnings should be emitted.
10. Test with a prerelease version of the extension that includes a MinimumChocolateyVersion attribute set to higher than 1.0.x and ensure that attempting to install Chocolatey in this scenario correctly also emits a warning (slightly different to the previous warning).
11. Check the installation still works on PowerShell v2.

## Change Types Made



## Related Issue

Fixes #2659

## Change Checklist
